populator-machinery/controller.go

```diff
@@ -527,13 +556,49 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 	}
 
 	// Look for the populator pod
-	podName := fmt.Sprintf("%s-%s", populatorPodPrefix, pvc.UID)
-	c.addNotification(key, "pod", c.populatorNamespace, podName)
 	var pod *corev1.Pod
-	pod, err = c.podLister.Pods(c.populatorNamespace).Get(podName)
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return err
+	podName := fmt.Sprintf("%s-%s", populatorPodPrefix, pvc.UID)
+	if c.usePod {
```

Comment:
Thinking a bit more this cannot be a flag at the controller layer. so some volume types we would need to go the no pod route, and for another volume type we would need to go the pod create route

Reply:
Discussed offline, the volume type specific logic should be implemented in the populate and populateComplete functions, since the type selected logic can not be implemented in the lib-volume-populator